entity_client.go

```diff
@@ -20,7 +20,7 @@ type EntityClient interface {
 	Insert([]Entity) ([]WriteResult, error)
 
 	// Update is a bulk operation that patches entities that match the query.
-	Update(query string, patch []Entity) ([]WriteResult, error)
+	Update(query string, patch Entity) ([]WriteResult, error)
```

Comment:
Good catch. Our original design was unclear if we wanted 1 or many patches, but 1 makes more sense. With many, handling failure becomes messy. E.g. if there are 3 patches and the last fails... is that ok? What's it mean to be 2/3rds patched?

Reply:
yeah, imo multiple patches doesn't make sense. i would either combine them into a single patch, or make multiple Update requests

Reply:
On second look, we _do_ want multiple patches. This is the nature of the bulk API endpoints: all ops in bulk: create, update, and delete. This is why `[]WriteResult` is returned, which also addresses my concern re partial failure. Create and delete already handle this, so update should too. So what really needs to be fixed is `entity.Store.UpdateEntities()`.

Reply:
Hm, on third look :-) this is still ambiguous. Atm, the API returns each old value ("diff") as the write result, but I'm not sure that's in the spirit of what `[]WriteResult` is was meant for. Also, is there any reason to do
```js
[
  {foo: "bar"},
  {x: 1}
]
```
and not just
```js
{foo:"bar", x: 1}
```
? After all, the patch can only match 1 query (or specific entity).

Reply:
Ok yeah, you're right, single patch is correct. The confusion is in the intent/usage of `[]WriteResult` but we'll address and fix that later because for now it works fine.